### PR TITLE
bpo-44794: Merge tests for typing.Callable and collection.abc.Callable

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue44794

### CODE REVIEW

File: Lib/test/test_typing.py [hunk 1]
```diff
@@ -4496,13 +4618,6 @@ class Z(Generic[P]):
         self.assertEqual(G5.__parameters__, G6.__parameters__)
         self.assertEqual(G5, G6)
 
-    def test_var_substitution(self):
-        T = TypeVar("T")
-        P = ParamSpec("P")
```
Comment (Fidget-Spinner) [inline Lib/test/test_typing.py:4501]: Where did this test go? I don't see a ParamSpec test in the new `test_var_substitution` test.

File: Lib/test/test_typing.py [hunk 2]
```diff
@@ -489,28 +497,142 @@ def f():
...
+
+    def test_consistency(self):
+        # bpo-42195
+        # Testing collections.abc.Callable's consistency with typing.Callable
+        c1 = typing.Callable[[int, str], dict]
+        c2 = collections.abc.Callable[[int, str], dict]
+        self.assertEqual(c1.__args__, c2.__args__)
+        self.assertEqual(hash(c1.__args__), hash(c2.__args__))
+
+    test_errors = skip("known bug #44793")(BaseCallableTests.test_errors)
```
Comment (Fidget-Spinner) [inline Lib/test/test_typing.py:629]: 👍 off topic: I find this really elegant.

File: Lib/_collections_abc.py [hunk 3]
```diff
@@ -478,8 +478,7 @@ def __getitem__(self, item):
         # then X[int, str] == X[[int, str]].
         param_len = len(self.__parameters__)
         if param_len == 0:
-            raise TypeError(f'There are no type or parameter specification'
-                            f'variables left in {self}')
+            raise TypeError(f'{self} is not a generic class')
```
Comment (Fidget-Spinner) [inline Lib/_collections_abc.py:481]: This is for consistency with the typing version right?

**Review by Fidget-Spinner — COMMENTED** [review]
LGTM. I just have a few questions below.

Phew, this refactored a lot of other tests in typing too. Thanks.

File: Lib/test/test_typing.py (hunk 1 above)
Comment (Fidget-Spinner) [reply Lib/test/test_typing.py:4501]: Ah nevermind, I found it: modified and moved to https://github.com/python/cpython/pull/27507/files#diff-04d29c98076c2d6bb75921ea9becb26a862544d39b71db87b6e354c759b9305dR580

File: Lib/_collections_abc.py (hunk 3 above)
Comment (serhiy-storchaka) [reply Lib/_collections_abc.py:481]: There was a minor bug in the old error message: missed space between "specification" and "variables".

Also, there is no reason to have different error messages for two implementations, so I changed one of them. I hope I left the better one.

Comment (Fidget-Spinner) [reply Lib/_collections_abc.py:481]: Oh I didn't notice the space. Good eye!

The message is copied from Py_GenericAlias for consistency with the rest of the collections.abc https://github.com/python/cpython/blob/main/Objects/genericaliasobject.c#L303

However, keeping it same with the old `typing.Callable` should be fine too.

**Pull request merged**